src/daemon/command_parser_executor.cpp

```diff
@@ -986,17 +986,30 @@ bool t_command_parser_executor::check_blockchain_pruning(const std::vector<std::
 
 bool t_command_parser_executor::set_bootstrap_daemon(const std::vector<std::string>& args)
 {
-  const size_t args_count = args.size();
-  if (args_count < 1 || args_count > 3)
+  size_t args_count = args.size();
+  if (args_count < 1 || args_count > 4)
   {
     std::cout << "Invalid syntax: Wrong number of parameters. For more details, use the help command." << std::endl;
     return true;
   }
 
+  std::string proxy;
+  {
+    const auto last_arg = args.rbegin();
+    constexpr const char proxy_prefix[] = "proxy=";
+    if (last_arg->find(proxy_prefix) == 0)
+    {
+      constexpr size_t proxy_prefix_length = sizeof(proxy_prefix) - sizeof(proxy_prefix[0]);
+      proxy = last_arg->substr(proxy_prefix_length);
+      --args_count;
+    }
+  }
```

Comment:
I think this will allow something like:
set_bootstrap_daemon a.b.c.d:2000   user pass foo

Also, I did not realize password was here, it should probably prompt, since you cna't otherwise have spaces etc (as a separate patch though, and not necessarily by you).

Reply:
Updated, reworked arguments parsing.